src/features/MovieDetails/movieDetailsSlice.js

```diff
@@ -0,0 +1,39 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const movieDetailsSlice = createSlice({
+    name: "movieDetails",
+    initialState: {
+        status: "initial",
+        movieData: null,
+        movieCredits: null,
+    },
+    reducers: {
+        fetchMovieDetails: () => ({
+            status: "loading",
+        }),
+        fetchMovieDetailsSuccess: (_, { payload: movieDetails }) => ({
+            status: "success",
+            movieData: movieDetails.movieData,
+            movieCredits: movieDetails.movieCredits,
+        }),
+        fetchMovieDetailsError: () => ({
+            status: "error",
+            movieData: null,
+            movieCredits: null,
+        }),
+    }
```

Comment:
Fun fact! 😵‍💫   We can write it this way as well: 
```
reducers: {
        fetchMovieDetails: state => {
            state.status = "loading";
        },
        fetchMovieDetailsSuccess: (state, { payload: movieDetails }) => {
            state.status = "success";
            state.movieData = movieDetails.movieData;
            state.movieCredits = movieDetails.movieCredits;
        },
        fetchMovieDetailsError: state => {
            state.status = "error";
            state.movieData = null;
            state.movieCredits = null;
        },
    }
```
I am not quite sure what's the difference but both ways work 😄

Reply:
interesting. I find the way You wrote it easier to read

Reply:
I think both is good. We just have to keep it consistent and use the same way in all our slices.